chisel/src/dispatcher.rs

```diff
@@ -0,0 +1,934 @@
+                        .iter()
+                        .map(|(time, name)| {
+                            format!("{} - {}", Paint::blue(format!("{:?}", time)), name)
+                        })
+                        .collect::<Vec<String>>()
+                        .join("\n")
+                ))),
+                Err(_) => DispatchResult::CommandFailed(Self::make_error(
+                    "No sessions found. Use the `!save` command to save a session.",
+                )),
+            },
+            ChiselCommand::Source => {
+                if let Some(session_source) = self.session.session_source.as_ref() {
+                    match format_source(
+                        &session_source.to_repl_source(),
+                        session_source.config.foundry_config.fmt.clone(),
+                    ) {
+                        Ok(formatted_source) => DispatchResult::CommandSuccess(Some(
+                            SolidityHelper::highlight(&formatted_source),
+                        )),
+                        Err(_) => DispatchResult::CommandFailed(String::from(
+                            "Failed to format session source",
+                        )),
+                    }
+                } else {
+                    DispatchResult::CommandFailed(Self::make_error("Session not present."))
+                }
+            }
+            ChiselCommand::ClearCache => match ChiselSession::clear_cache() {
+                Ok(_) => {
+                    self.session.id = None;
+                    DispatchResult::CommandSuccess(Some(String::from("Cleared chisel cache!")))
+                }
+                Err(_) => DispatchResult::CommandFailed(Self::make_error("Failed to clear cache!")),
+            },
+            ChiselCommand::Fork => {
+                if let Some(session_source) = self.session.session_source.as_mut() {
+                    if args.is_empty() {
+                        session_source.config.evm_opts.fork_url = None;
+                        return DispatchResult::CommandSuccess(Some(String::from(
+                            "Now using local environment.",
+                        )))
+                    } else if args.len() != 1 || args[0].trim().is_empty() {
+                        return DispatchResult::CommandFailed(Self::make_error(
+                            "Must supply a session ID as the argument.",
+                        ))
+                    }
+
+                    // If the argument is an RPC alias designated in the
+                    // `[rpc_endpoints]` section of the `foundry.toml` within
+                    // the pwd, use the URL matched to the key.
+                    let endpoint = if let Some(endpoint) =
+                        session_source.config.foundry_config.rpc_endpoints.get(args[0])
+                    {
+                        endpoint.clone()
+                    } else {
+                        RpcEndpoint::Env(args[0].to_owned())
```

Comment:
nit: convert str::to_owned into str::to_string

there's no difference but makes it consistent across (most of) the codebase
```suggestion
                        RpcEndpoint::Env(args[0].to_owned())
```